airflow/models/dag.py

```diff
@@ -2403,6 +2507,24 @@ def __init__(self, concurrency=None, **kwargs):
     def __repr__(self):
         return f"<DAG: {self.dag_id}>"
 
+    @property
+    def data_interval(self) -> Optional[Tuple[datetime, datetime]]:
+        if self.next_dagrun_data_interval_start is None:
+            if self.next_dagrun_data_interval_end is not None:
+                raise AirflowException(
+                    "Inconsistent DagModel: next_dagrun_data_interval_start and "
+                    "next_dagrun_data_interval_end must be either both None or both datetime"
+                )
+            return None
+        return (self.next_dagrun_data_interval_start, self.next_dagrun_data_interval_end)
```

Comment:
This seems... odd. What does `dag.data_interval` _mean_? I would expect this at least to be called `next_data_interval` based on the implementation.

Reply:
Yes, this should be `next_data_interval` instead.